src/hal/user_comps/xhc-whb04b-6/hal.cc

```diff
@@ -1427,29 +1452,45 @@ bool Hal::waitForRequestedMode(volatile hal_bit_t * condition)
     return false;
 }
 // ----------------------------------------------------------------------
+void Hal::monitorResetState() {
+    if (*memory->out.spindleOverrideDoIncrease && expired(mToggleSpindleOverrideIncreaseStamp))
+            *memory->out.spindleOverrideDoIncrease=false;
+    if (*memory->out.spindleOverrideDoDecrease && expired(mToggleSpindleOverrideDecreaseStamp))
+            *memory->out.spindleOverrideDoDecrease=false;
+}
+
+bool Hal::expired(int64_t t) {
+        int64_t timestamp;
+        timestamp = now();
+        if (t<timestamp)
+                return true;
+        else
+                return false;
+}
+
+int64_t Hal::now() {
+        int64_t timestamp;
+        struct timespec ts;
+        clock_gettime(CLOCK_MONOTONIC, &ts);
+        timestamp = (int64_t)(ts.tv_sec*1000)+ts.tv_nsec/1000000;
+        return timestamp;
+}
+
```

Comment:
Inconsistent indentation.  Should be four spaces per level: http://linuxcnc.org/docs/2.8/html/code/style-guide.html#_indentation